src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -15,7 +14,7 @@ namespace Microsoft.Extensions.DependencyInjection
     /// <summary>
     /// The default IServiceProvider.
     /// </summary>
-    internal class ServiceProvider : IServiceProvider, IDisposable
+    public sealed class ServiceProvider : IServiceProvider, IDisposable
```

Comment:
The other reason I made it sealed is to avoid things like https://github.com/aspnet/DependencyInjection/issues/476